### DISCUSSION TITLE
dotnet lambda package - AOT - Container- How to pass Nuget.config file as parameter

### DISCUSSION THREAD

**Discussion by rajeshaz09** [aws/aws-extensions-for-dotnet-cli#263] [Q&A] [answered]
My lambda project references nuget packages that are there in our company's private nuget repository. I created Nuget.config with my repository as additional source. 

I used following dotnet commands in CI/CD pipeline to genarate lambda package

- dotnet resotre --configfile <nuget.config location>
- dotnet lambda package

this setup is working fine.  

But when i enable AOT, as specified in the documentation, AL2 container launched to create lambda package. 
I received error saying unable to restore xxxxx package from nuget.org. 

My question is how to pass Nuget.config file location to "dotnet lambda package" command so that it will internally pass to AL2 container during restore/publish? 

**Environment details**

- Dotnet - 7.0
- Architecture - Linux-x64 - AOT enabled

**Comment by ashishdhingra** [comment, marked as answer]
@rajeshaz09 Good morning. While there is no dedicated argument for specifying NuGet source, the AWS Extensions for .NET CLI supports [--publish-options](https://github.com/aws/aws-extensions-for-dotnet-cli/blob/d82c052e04e888a4592bf766273a80d811d94ed5/src/Amazon.Common.DotNetCli.Tools/Options/CommonDefinedCommandOptions.cs#L36) that could be used to pass additional parameters to `dotnet publish` command. The [dotnet publish](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-publish) command supports `--source` parameter which specifies the `URI of the NuGet package source to use during the restore operation`. You try passing value for `--publish-options` as `"--source <<URI>>"` to use your NuGet source.

Thanks,
Ashish

[a comment by an automated account is not shown]
